gson/src/test/java/com/google/gson/JsonPrimitiveTest.java

```diff
@@ -169,6 +169,25 @@ public void testLongEqualsBigInteger() {
     assertEquals(p1.hashCode(), p2.hashCode());
   }
 
+  public void testHashCodeBigIntegerEqualsBigInteger() {
+    // within long range
+    BigInteger smallBigInt = new BigInteger("10");
+    JsonPrimitive p1 = new JsonPrimitive(smallBigInt);
+    JsonPrimitive p2 = new JsonPrimitive(smallBigInt);
+    assertTrue("should be different objects: " + p1 + ", " + p2, p1 != p2);
```

Comment:
Not needed because above `p1` and `p2` are assigned two new (therefore not-same) instances